docs/configuration/configuration.md

```diff
@@ -1,87 +1,166 @@
+Environment variable keys are uppercase, prefixed with `FLAGD_` and all `-` are replaced with `_`. For example,
+`metrics-port` flag in environment variable form is `FLAGD_METRICS_PORT`.
 
-## URI patterns
+Config file expects the keys to have the exact naming as startup-flags flags.
 
-Any URI passed to flagd via the `--uri` flag must follow one of the 4 following patterns to ensure that it is passed to the correct implementation:
+## Sync providers
 
-| Sync       | Pattern                               | Example                               |
-|------------|---------------------------------------|---------------------------------------|
-| Kubernetes | `core.openfeature.dev/namespace/name` | `core.openfeature.dev/default/my-crd` |
-| Filepath   | `file:path/to/my/flag`                | `file:etc/flagd/my-flags.json`        |
-| Remote     | `http(s)://flag-source-url`           | `https://my-flags.com/flags`          |
-| Grpc       | `grpc(s)://flag-source-url`           | `grpc://my-flags-server`              |
+Sync providers are a core part of flagd. They are the sources of feature flag configurations for evaluations. flagd
+currently support the following sync providers,
 
-## Customising sync providers
-
-Custom sync providers can be used to provide flag evaluation logic.
+* [Kubernetes provider](#kubernetes-provider)
+* [Filepath Configuration](#filepath-provider)
+* [Remote Configuration](#remote-provider)
+* [GRPC Configuration](#grpc-provider)
 
 ### Kubernetes provider
 
-The Kubernetes provider allows flagD to connect to a Kubernetes cluster and evaluate flags against a specified FeatureFlagConfiguration resource as defined within the [open-feature-operator](https://github.com/open-feature/open-feature-operator/blob/main/apis/core/v1alpha1/featureflagconfiguration_types.go) spec.
+The Kubernetes sync provider allows flagD to connect to a Kubernetes cluster and evaluate flags against a specified
+FeatureFlagConfiguration resource as defined within
+the [open-feature-operator](https://github.com/open-feature/open-feature-operator/blob/main/apis/core/v1alpha1/featureflagconfiguration_types.go)
+spec.
 
 To use an existing FeatureFlagConfiguration custom resource, start flagD with the following command:
 
 ```shell
 flagd start --uri core.openfeature.dev/default/my_example
 ```
 
-## Source Configuration
+In this example, `default/my_example` expected to be a valid FeatureFlagConfiguration resource, where `default` is the
+namespace and `my_example` being the resource name.
+
+### Filepath provider
+
+File path sync provider reads and watch the source file for updates(ex:- changes and deletions).
+
+```shell
+flagd start --uri file:etc/featureflags.json
+```
+
+In this example, `etc/featureflags.json` is a valid feature flag configuration file accessible by the flagd runtime.
+
+### Remote provider
+
+Remote sync provider fetch flags from a remote source and periodically poll the source for flag configuration updates.
```

Comment:
```suggestion
The HTTP sync provider fetch flags from a remote source and periodically poll the source for flag configuration updates.
```

We've generally tried to replace "remote" to "http". "grpc" is also possibly "remote" so "http" is a better name.